packages/design-system/src/components/primitives/is-truncated.tsx

```diff
@@ -0,0 +1,18 @@
+import { type RefObject, useState, useEffect } from "react";
+
+/**
+ * Checks whether text was truncated in an element with `text-overflow: ellipsis`
+ */
+export const useIsTruncated = (ref: RefObject<HTMLElement>, text: string) => {
+  const [isTruncated, setIsTruncated] = useState(false);
+
+  useEffect(() => {
+    if (ref.current) {
+      const { offsetWidth, scrollWidth } = ref.current;
+      // https://stackoverflow.com/a/10017343/478603
+      setIsTruncated(offsetWidth < scrollWidth);
+    }
+  }, [ref, text]);
```

Comment:
ref doesn't change

Reply:
usually if you need a ref in the deps its a smell

Reply:
you could use ref function instead, so you can know if element has changed and then set the truncated

Reply:
Yes. It doesn't change. Having it in deps makes no difference, but if I remove eslint will complain.

Reply:
Added a comment